src/System.IO.Pipelines/PipelineConnectionExtensions.cs

```diff
@@ -81,41 +81,54 @@ public static Task CopyToEndAsync(this IPipeReader input, Stream stream)
         {
             try
             {
-                await input.CopyToAsync(stream, bufferSize, cancellationToken);
+                // TODO: Use bufferSize argument
+                while (!cancellationToken.IsCancellationRequested)
+                {
+                    var result = await input.ReadAsync();
+                    var inputBuffer = result.Buffer;
+                    try
+                    {
+                        if (inputBuffer.IsEmpty && result.IsCompleted)
+                        {
+                            input.Complete();
+                            return;
+                        }
+
+                        await inputBuffer.CopyToAsync(stream);
+                    }
+                    finally
+                    {
+                        input.Advance(inputBuffer.End);
+                    }
+                }
             }
             catch (Exception ex)
```

Comment:
I'm not sure it makes as much sense to swallow exceptions when copying to a Stream since the implementer of the stream can't observe the exception when IPipeReader.ReadAsync() throws.

There's also a decent chance the producer won't observe exceptions from Stream.WriteAsync() either if it throws after all the produced data has already been "flushed".

What do you think @davidfowl?